### PR TITLE
Descriptor expansion cache clarifications

### PR DESCRIPTION
**Pull request by Sjors**
I found the name `m_script_arg` to be confusing while reviewing https://github.com/bitcoin/bitcoin/pull/14646#discussion_r240677238. @sipa let me know if `m_subdescriptor_arg` is completely wrong.

I also added an explanation of why we call `GetPubKey` when we don't ask it for a public key.

### CODE REVIEW

**Comment by fanquake** [comment]
Might as well fixup the other nit (trivial) [nits](https://github.com/bitcoin/bitcoin/pull/14646#pullrequestreview-183028560).

**Comment by Sjors** [comment]
@fanquake done, I snuck those into the second commit.

File: src/script/descriptor.cpp [hunk 1]
```diff
@@ -295,6 +295,8 @@ class DescriptorImpl : public Descriptor
         // Construct temporary data in `entries` and `subscripts`, to avoid producing output in case of failure.
         for (const auto& p : m_pubkey_args) {
             entries.emplace_back();
+            // If we have a cache, GetPubKey should not actually return a public key,
+            // so we pass a nullprt. We're only interested in uncached origin info.
```
Comment (flack) [inline src/script/descriptor.cpp]: Typo: `nullprt`

Comment (Sjors) [reply src/script/descriptor.cpp]: Fixed

**Comment by MeshCollider** [comment]
`m_script_arg` refers to the defined `SCRIPT` in the descriptor doc, to make it distinct from `KEY` or `ADDR`. But subdescriptor is also a clear name so I have no strong preference either way.

File: src/script/descriptor.cpp [hunk 2]
```diff
@@ -295,6 +295,8 @@ class DescriptorImpl : public Descriptor
         // Construct temporary data in `entries` and `subscripts`, to avoid producing output in case of failure.
         for (const auto& p : m_pubkey_args) {
             entries.emplace_back();
+            // If we have a cache, GetPubKey should not actually return a public key,
```
Comment (sipa) [inline src/script/descriptor.cpp]: "should not" sounds like it should be prohibited from computing the pubkey.

What about "If we have a cache, we don't need GetPubKey to compute the public key. Pass in nullptr to signify only origin info is desired.".

Comment (Sjors) [reply src/script/descriptor.cpp]: Done

**Comment by Sjors** [comment]
Rebased.

@MeshCollider I added a comment referencing `SCRIPT` in the descriptor docs.

By the way, I noticed that neither `DescriptorImpl` nor `m_subdescriptor_arg` show up in the docs, not even in the docs for `script/descriptor.cpp`. I don't understand Doxygen well enough to know if that's a feature (i.e. because these functions are private).

**Comment by Sjors** [comment]
Miraculously this doesn't need another rebase...

**Comment by practicalswift** [comment]
FWIW: I've verified that a disassembly of the `bitcoind` binary built with this patch applied is identical to a disassembly of the `bitcoind` binary built against `master` (as expected).

**Comment by laanwj** [comment]
ACK 2e68ffaf205866e4cea71f64e79bbfb89e17280a

**Pull request merged**